web/src/app/admin/admin-message-logs/AdminDebugMessagesLayout.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react'
+import { useQuery, gql } from '@apollo/client'
+import { Grid, Typography } from '@mui/material'
+import makeStyles from '@mui/styles/makeStyles'
+import { GenericError } from '../../error-pages'
+import Spinner from '../../loading/components/Spinner'
+import DebugMessagesList from './DebugMessagesList'
+import DebugMessagesControls from './DebugMessagesControls'
+import DebugMessageDetails from './DebugMessageDetails'
+import { theme } from '../../mui'
+import { DebugMessage } from '../../../schema'
+
+export const MAX_QUERY_ITEMS_COUNT = 1000
+const LOAD_AMOUNT = 50
+
+const debugMessageLogsQuery = gql`
+  query debugMessageLogsQuery($first: Int!) {
+    debugMessages(input: { first: $first }) {
+      id
+      createdAt
+      updatedAt
+      type
+      status
+      userID
+      userName
+      source
+      destination
+      serviceID
+      serviceName
+      alertID
+      providerID
+    }
+  }
+`
+
+const useStyles = makeStyles<typeof theme>((theme) => ({
+  containerDefault: {
+    [theme.breakpoints.up('md')]: {
+      maxWidth: '100%',
+      transition: `max-width ${theme.transitions.duration.leavingScreen}ms ease`,
+    },
+  },
+  containerSelected: {
+    [theme.breakpoints.up('md')]: {
+      maxWidth: '70%',
+      transition: `max-width ${theme.transitions.duration.enteringScreen}ms ease`,
+    },
+  },
+  groupTitle: {
+    fontSize: '1.1rem',
+  },
+  saveDisabled: {
+    color: 'rgba(255, 255, 255, 0.5)',
+  },
+  card: {
+    margin: theme.spacing(1),
+    cursor: 'pointer',
+  },
+  textField: {
+    backgroundColor: 'white',
+    borderRadius: '4px',
+    minWidth: 250,
+  },
+}))
+
+export default function AdminDebugMessagesLayout(): JSX.Element {
+  const classes = useStyles()
+
+  // all data is fetched on page load, but the number of logs rendered is limited
+  const [numRendered, setNumRendered] = useState(LOAD_AMOUNT)
+  const [selectedLog, setSelectedLog] = useState<DebugMessage | null>(null)
```

Comment:
Would it be possible to take the URL params here and filter the items we get from data? It seems like we're getting the data here, passing it down, and then asking for events & info back from filtering up the tree causing multiple renders and making the data flow more complicated than necessary.

Maybe if we use fuse here, and just pass the total/filtered/showing counts down we can bypass that. For change events we can pass event handlers to `DebugMessageControls`

Maybe something like this to decouple things a bit:
```jsx
<DebugMessageControls search={search} onSearchChange={setSearch} />
```

Then `DebugMessagesList` can just render the messages we pass to it and have a couple of props for "load more" events but all the processing/control will happen here in the component in charge of getting the data.

Reply:
I looked into this when making the PR, but moving the hooks up to the parent component would be difficult since that parent renders conditionally depending on network data. The solution would be to add another component in-between, so the structure would look like this:
```
             AdminDebugMessagesLayout (Network requests & conditionally render page)
                                                               |
                AdminDebugMessages (Hold state and do all the filtering/limiting/sorting)
                         /                                     |                                \
DebugMessageControls  DebugMessagesList   DebugMessageDetails (Mostly stateless children components)
```
I decided against it just because it wasn't totally necessary and we could get this PR out faster. We did plan to make a V2 PR though to clean up some things, and we can make this change in there.

Reply:
The idea in the current implementation was that the controls handles all filtering/searching/controlling what is shown in the list, and the layout would set the page layout based on loading/not. I can see how having to call the URL param hooks more than once can get cumbersome, but overall I think it makes sense rather than doing everything inside of the Layout file and bogging it down- since it would feel awkward to pass down the search/setSearch as props when we can just grab it with the `useURLParam` hook wherever we need it. I agree though that maybe a HOC in as a middle layer can control it all instead, if there is an issue with the page re-rendering too many times. That would bring the setSearch logic closer to the filtering logic as well.

Reply:
Layout (queries for data then render loading/error/data)
↪ DebugMessages (controls all the data to filter based on URL Params, as well as the selected log)
↪ Controls (setters passed down to buttons) ↔ List (renders data) ↔ Details (renders more data based on selected log)